sklearn/linear_model/_ridge.py

```diff
@@ -1616,6 +1616,18 @@ def fit(self, X, y, sample_weight=None):
             self.alpha_ = gs.best_estimator_.alpha
             self.best_score_ = gs.best_score_
 
+        min_alpha = min(self.alphas)
+        max_alpha = max(self.alphas)
+        if self.alpha_ in [min_alpha, max_alpha]:
+            warnings.warn(f"The optimal value for the regularization "
+                          "parameter 'alpha' was {self.alpha_}\n which "
+                          "lies at a boundary of the explored range "
+                          "(between {min_alpha} and {max_alpha}).\n Consider "
+                          "setting the 'alphas' parameter to explore a "
+                          "wider range.\n To ignore warnings try running "
+                          "\nimport warnings "
+                          "\nwarnings.filterwarnings('ignore')")
```

Comment:
If we do a usual `UserWarning` as in most other places in the library, then I suppose we don't need to have this (as @jnothman said ready). You can remove this part then.

Reply:
Okie @adrinjalali I will a add a test as well! : )

Reply:
Okie @adrinjalali I will add a test. Then I will also extend this to LogisticregressionCV : )